### DISCUSSION TITLE
"Magic" way to avoid double navigation on links [up-follow]

### DISCUSSION THREAD

**Discussion by alexandremjacques** [unpoly/unpoly#561] [Q&A] [answered]
Hi,

Quick question: is there an Unpoly way to block user from clicking more then once on the same link? As I mentioned earlier, a slow loading page "makes" users click more than once on the same link (I have the loading bar feedback but its not enough, apparently).

I was thinking of using a full-screen loading overlay to block user from clicking mtultiple times. Can Unpoly help me with this approach? Any other ideas?

TIA

**Comment by triskweline** [comment, marked as answer]
By default we have [`[up-disable]`](https://unpoly.com/form-up-submit#up-disable), but that only works for forms.

You could convert your link into a form button like this:

```html
<form method="get" action="/report" up-submit up-disable>
  <button>Open report</button>
</form>
```

> I was thinking of using a full-screen loading overlay to block user from clicking mtultiple times. 

If it needs to be a link, maybe something like this would work:

```html
<a href="/report" class="slow-link">Open report</a>
```

```js
function showOverlay() { ... }
function hideOverlay() { ... }

up.on('a.slow-link', 'up:link:follow', ({ renderOptions }) => {
  showOverlay()
  renderOptions.onLoaded = hideOverlay
  renderOptions.onError = hideOverlay
})
```

**Reply by altla** [reply]
Any plans to add anchor tag support to up-disable? I was looking for this exact thing today.

Converting to a form isn’t always ideal because you can’t make them preload.

(I will try your JavaScript solution though. Thanks!)

**Reply by triskweline** [reply]
> Any plans to add anchor tag support to up-disable?

I think this would be a nice addition.

**Comment by alexandremjacques** [comment]
That's great! Thanks a lot!
